### PR TITLE
Add new isPasskeyPlatformAuthenticatorAvailable() method

### PR DESCRIPTION
**Pull request by timcappalli**
The current developer guidance to relying parties for whether they can offer passkeys to users is to call `isUVPAA()` and `isConditionalMediationAvailable()`. One returns a boolean, the other a promise. This is already complex . This also doesn't tell the whole story. For example, in Chrome and Edge on Ubuntu, users can currently create a syncable passkey on another device such as a phone or tablet using Cross-Device Authentication (hybrid transport). There is no way for a relying party to know this today, and passkeys won't be offered to users. This is a similar case with smart displays and other devices such as kiosks.

This method is designed to be a very simple way for the majority of developers to detect whether they can offer passkeys to users.

For more advanced use cases, `isUVPAA()` remains available, and an upcoming PR will introduce a more advanced feature detection method for things like specific transports (e.g. is just hybrid available) and CTAP2 security keys with client PIN entry support.

This was originally proposed during the WebAuthn F2F in April 2023.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1901.html" title="Last updated on Jun 28, 2023, 2:13 PM UTC (8813987)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1901/ad61c6f...8813987.html" title="Last updated on Jun 28, 2023, 2:13 PM UTC (8813987)">Diff</a>

### CODE REVIEW

File: index.bs
Comment (emlun) [inline index.bs:2769]: I think?
```suggestion
If neither is discovered, the promise is resolved with the value of [FALSE].
```

File: index.bs
Comment (emlun) [inline index.bs:4376]: The prose below also needs to be updated - in particular the "The four combinations not named in [the table]" list, where the first item needs to be removed or updated (and the "four" needs to change too - we should probably just delete it and just say "The combinations not named [...]". I don't think it's ever been 4, actually.)

File: index.bs
Comment (emlun) [inline index.bs:4369]: If we're introducing the "passkey" term, should we also rename "first-factor roaming authenticator" to "passkey roaming authenticator"? The "first-factor" term has never really caught on and is awkwardly mismatched with "user-verifying platform authenticator", and there's only one use of it outside the Authenticator Taxonomy section.

File: index.bs
Comment (emlun) [inline index.bs:4319]: Above this there's this example:

>When authenticating for the first time on a particular [client device](https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1901.html#client-device), a [roaming authenticator](https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1901.html#roaming-authenticators) is typically needed since the user doesn’t yet have a [platform credential](https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1901.html#platform-credential) on that [client device](https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1901.html#client-device).

which is probably no longer accurate with synced passkeys.

File: index.bs
Comment (emlun) [reply index.bs:4369]: On a similar train of thought: should we remove the "User-verifying platform authenticator" term from this list? It's largely superseded by the passkey class. We could demote it to the below list of less interesting combinations, since it's unlikely that a platform authenticator would be multi-factor capable but not client-side-storage capable.

Comment (Kieun) [reply index.bs:4369]: What is the `passkey platform authenticator`? We have passkey, platform authenticator and cross-platform authenticator. For me, the term is quite vague. Is it better to introduce syncable, multi-device or something?
And we have some of 3rd party passkey providers as well (which is not the platform authenticator). I think it's time to re-define the authenticator taxonomy.

Comment (timcappalli) [reply index.bs:4369]: > And we have some of 3rd party passkey providers as well (which is not the platform authenticator). 

3P passkey providers are considered the same as 1P passkey providers and can be both platform and cross-platform, which is what this change reflects.

File: index.bs
Comment (timcappalli) [reply index.bs:2769]: Need a third person to do a grammar check. "Is" doesn't sound correct to me. @MasterKale?

File: index.bs
Comment (timcappalli) [reply index.bs:4319]: Yes, I think this whole section needs some updating, but I was only trying to address changes specific to the new method.

File: index.bs
Comment (MasterKale) [reply index.bs:2769]: So the **English Language & Usage** Stack Exchange claims that "neither is" is _technically_ correct, but "neither are" is more common. Do specs like ours aim for technical correctness is all aspects, or are colloquialisms allowed in?

Personally I say "neither are" because it sounds more natural, but if it's for sake of "linguistic clarity" across the board I'd say go with "neither is".

https://english.stackexchange.com/questions/10304/which-is-correct-neither-is-or-neither-are

**Comment by MasterKale** [comment]
I'm happy for "passkey" to simply be made equivalent in WebAuthn to a "discoverable credential". The sync-ability of discoverable credentials is neither here nor there for the spec, which simplifies things greatly.

Regarding method name, I think `isPasskeyPlatformAuthenticatorAvailable()` ("isPPAA" 🤔) is fine. It's still too long for my personal preference, but because it follows the "PlatformAuthenticatorAvailable" pattern we've got with isUVPAA (and I can't think of a more succinct name for a method like this) then I'd be willing to commit to this one as proposed.

> > `isPlatformBackupEligibleAuthenticatorAvailable()` would be more precise.
> 
> @ndpar that is not what the method is conveying. This PR contains a definition of passkeys, aliased to the existing definition of a discoverable credential.

@ndpar I think I have to agree with Tim on this one, there's of course nothing about a discoverable credential that would prevent one from being used in a hybrid auth flow if it wasn't synced, nor via local platform authenticator under the same conditions. Thus I too believe it's better to leave out the idea of backup eligibility from this method, since all it's trying to establish is a developer-friendly method of intuiting if the conditions are right for passkey auth to likely succeed.

And from the RP's perspective I like that this presents a simple initial feature check that the browser is in a position to do hybrid registration/auth (it's currently impossible, for example, for an RP to understand via JS that a hybrid auth attempt failed because the browser didn't have Bluetooth permissions at the OS level).

I _might_ argue that I'd prefer the check for a client's hybrid prerequisites being satisfied to be a separate API method, then devs could simply chain the existing isUVPAA method with such a hybrid check method. But if we get such a capability to check for that as part of a more full-featured feature detection API somewhere down the line then I think I'd be fine with this method combining the two signals.

**Comment by ndpar** [comment]
You guys just proved my point that the word "passkey" is confusing :) It must be defined precisely if you want to use it in the technical documentation. And I'm afraid your definition will differ from what most people think when they hear word "passkey", so I would avoid introducing it to the spec, unless you are ready to promote and defend your definition across the board :)

**Comment by Firstyear** [comment]
> > `isPlatformBackupEligibleAuthenticatorAvailable()` would be more precise.
> 
> @ndpar that is not what the method is conveying. This PR contains a definition of passkeys, aliased to the existing definition of a discoverable credential.

ndpar is correct though - despite what the standards authors may desire, there are at least 4 different definitions of what a passkey is in use by vendors.

* Any possible webauthn credentials (cite https://freeipa.readthedocs.io/en/latest/designs/passkeys.html )
* A synchronised/roaming credentials (cite https://support.okta.com/help/s/article/Passkey-Management?language=en_US )
* A credential that has BE/BS=true - I can't find an obvious cite for this, android could fall into this category though in a way.
* And the passkeys are resident keys as I'm sure you're aware of.

As a result, I think it's not correct for us to use an ambiguous and murky marketing term in a specification.

**Review by Firstyear — CHANGES_REQUESTED** [review]
Marking as needing changes due to passkey being ambiguous.

**Comment by timcappalli** [comment]
> ndpar is correct though - despite what the standards authors may desire, there are at least 4 different definitions of what a passkey is in use by vendors.

We're not referring to external definitions. We're referring to the definition in the specification.

Not sure what you mean by marketing term. Is password also a marketing term? Because password is used throughout Credential Management, the parent spec to WebAuthn.

**Comment by MasterKale** [comment]
> You guys just proved my point that the word "passkey" is confusing :) It must be defined precisely if you want to use it in the technical documentation. And I'm afraid your definition will differ from what most people think when they hear word "passkey", so I would avoid introducing it to the spec, unless you are ready to promote and defend your definition across the board :)

But this diff is attempting to codify the meaning of a passkey in the context of WebAuthn: it's a [Discoverable Credential](https://w3c.github.io/webauthn/#discoverable-credential). Nothing more nor less than what is currently included in L3. If "discoverable credential" is confusing then that means the spec needs clarification about what a discoverable credential is.

**Review by MasterKale — APPROVED** [review]

**Comment by Firstyear** [comment]
> But this diff is attempting to codify the meaning of a passkey in the context of WebAuthn: it's a [Discoverable Credential](https://w3c.github.io/webauthn/#discoverable-credential). Nothing more nor less than what is currently included in L3. If "discoverable credential" is confusing then that means the spec needs clarification about what a discoverable credential is.

The only definition that matters is what people are using in reality - and sadly as much as we can define it here as many times as we want, the cat is out of the bag. Passkeys are a vague term and I think it can't be undone at this point.

**Comment by timcappalli** [comment]
> The only definition that matters is what people are using in reality - and sadly as much as we can define it here as many times as we want, the cat is out of the bag. Passkeys are a vague term and I think it can't be undone at this point.

Which is why we add a specific definition in the context of the specification, that can be referenced throughout the spec when the term is used. Part of this PR does that.

**Comment by Firstyear** [comment]
> Which is why we add a specific definition in the context of the specification, that can be referenced throughout the specification whenever the term is used. Part of this PR does just that, in support of the new method defined.

I think what you're saying is "If we define it here, that will fix it, so everyone will use the term in the same consistent way".

I'm saying "People are already using the term in a non-consistent way, and they will never change that behaviour no matter how much we define it".

In fact, by trying to add the term "passkey" to the webauthn specification it may in fact make the situation worse as people will arrive at the specification with a preconceived notion of what a passkey is and that may lead to misinterpretation of the specification due to this.

I think it's inappropriate to add a marketing term into a specification retroactively when it's already being used in various inconsistent ways in the public. 

That's all I have to say on the matter, I'll leave it to others to agree or disagree with this.

**Comment by MasterKale** [comment]
> Are we deprecating it and replacing with "passkey"?

Based on the contents of this PR I wouldn't say anyone is proposing this. I'd characterize adding "passkey" as an alias of "discoverable credential" as an enabler of nicer API design. If `isPasskeyPlatformAuthenticatorAvailable()` is unacceptable because it introduces "passkey" into the spec, then we're stuck entertaining laughably long method names like `isDiscoverableCredentialPlatformAuthenticatorAvailable()`. Have we learned nothing from isUVPAA? 😂

**Comment by Firstyear** [comment]
> isDiscoverableCredentialPlatformAuthenticatorAvailable

Consistency and clarity is more important - at least this name for all it's verbosity is direct about exactly what is requested. Where as isPasskeyPlatformAuthenticatorAvailable could mean "any possible authenticator", "a synced credential", "a credential capable of backups" or "discoverable credentials".

**Comment by ko-koiwai** [comment]
Agree with @Firstyear. My understanding is that FIDO Alliance officially adopted that "passkey" is any form of FIDO credentials even including UAF U2F (pls ask your borad representatives). As FIDO2 is a joint effort of W3C and FIDO Alliance, it is not developer friendly to use the same word in different ways in two organizations.

**Comment by arianvp** [comment]
I personally think linking the term `Passkey` to `Discoverable Credential` is artificially limiting and probably a mistake. For example it would be incompatible with how Google currently categorises Passkeys (My non-discoverable yubikey credentials were upgraded from "security key" to "Passkey" in their interface and they work fine for Passwordless auth even though they're not stored as a resident credential).

Passkeys are about passwordless authentication to me.  Usernameless is nice; but this flow is not widely adopted. 

Tying it to Discoverable Credentials causes more confusion than needed in my opinion.  Is my passkey I have for my Google Account not a Passkey anymore? Why not? It works fine for log in.

**Comment by ve7jtb** [comment]
If we step back from the name discussion slightly, what are we trying to help the RP decide?

1) can the browser/platform the user is currently on make a credential that will work in my loginflow?
   This is tricky because there is more than one type of login flow.  Some require discoverable credentials.  
Others may require UV but we can probably safely assume that most authenticators capable of discoverable credentials are also capable of UV even if they prefer not to do it by default.

2) If a user has registered a credential should they provide conditional mediation or a login with passkey button.
I suspect that conditional mediation should be done on any browser that supports it every time.   
For the login with passkey button that is tricky before knowing who the user is and if they have credentials registered.

Without knowing the user what info is actually useful to a RP? 
Are there any browser/platforms that support WebAuthn but don't have a useful CAP2/platform authenticator option?

1) Windows 
     Win 11 is all good all browsers that I know of support platform and CTAP2 roaming authenticators.  All browsers other than Fire Fox support Hybrid.
     Win 10 other than some early releases the story is the same as win 11
     Win 7 and early Wun 10 support external authenticators and Hybrid on Chrome (and Edge if installed).  Fire Fox supports  CTAP2.

On Windows the combination of isUVPAA, conditional Mediation and FF CTAP2 support flag probably tells the RP everything useful.

2) Android 
    Support for platform credentials is there, but CTAP2 support is still being developed. It also doesn't support Hybrid out to another device like an iPhone.  So despite being a bit broken I think RP would want to show the passley login option on Android

3) iOS /iPadOS Safari and the other browsers use the built in API and support everything including Hybrid out.  So Gold star for most consistent platform, if only it could set a pin:)

4) MacOS Safari is all good, Chrome is all good (will improve with native UVPAA), Firefox has CTAP2 but no Hybrid or UVPAA until integrating with Apple's API.
So no reason for RP to block anything with the possible exception FF for the moment.

5) Linux. Chrome and FF have CTAP2 support and Chrome has Hybrid.  UVPAA is being developed by some people but your guess is as good as mine.   

So most browser/platforms shouldn't be blocked.  
Which ones do RP not want to use and why?  
Can those be identified by existing mechanisms?

I have the feeling that other than with Linux these are all short term problems. 

An indication that Hybrid is supported might be useful but can be determined with a bit of logic by the RP.

The thing with Hybrid is that you can't know if the user has another device, so it is at best a hint.
There are no browser platforms that don't support some sort of discoverable credential.

Lets agree on the property we are trying to identify with this.   

Is it that the Hybrid transport is possible, or that one of UVPAA or Hybrid is possible?

I think at this point the discoverable or not question is probably not useful as everything will reply yes.

Once we all agree on the property and that it is useful and not just creating more confusion then we can pick a name.

John B.
I might go with something like isUVPAA+ to show that there is a possibility of UVPAA but that it may be on a different physical device, and move to deprecate isUVPAA over time.

**Comment by MasterKale** [comment]
> Is it that the Hybrid transport is possible, or that one of UVPAA or Hybrid is possible?

The aspect of this PR that is most interesting to me as an RP is having some kind of signal that hybrid auth is possible. Right now auth can appear feasible (WebAuthn is present and isUVPAA returns true, for a user with an `["internal", "hybrid"]` credential), but auth fails because Chrome wants to prompt for hybrid but can't because the user clicked No weeks ago when Chrome unexpectedly prompted for Bluetooth permission.

We of course can't know via JavaScript that Chrome is missing prerequisites at the OS level for hybrid, so an API to preemptively know that hybrid is possible (a simple true/false that's really the browser telling the RP, "yes, I programmatically have everything I need to do hybrid" without getting into specifics) feels innocuous enough to me for me to be hopeful that we can work something like this into the spec.

That's my motivation for approving this PR. But I wonder too if this concept might also spur more discussion around a way for an RP to understand what ceremonies the client can handle, to tailor communications for the WebAuthn registration and authentication ceremonies for the end user's sake.

(This also ties a bit into earlier conversations we've had around how an RP could hint that it wants to do hybrid registration instead of security key registration for a `"cross-platform"` attachment.)

**Comment by ve7jtb** [comment]
Are browsers really going to set this flag to false if they have a platform authenticator and support Hybrid but some setting in the OS is stopping hybrid from being used? 

I don't think I got that.  

That is an interesting question,  I guess on Windows this could happen but I think platform support for Hybrid is coming in the next several months so it may be a short term problem on Windows.    On MacOS there is platform support for Hybrid that will get around the problem if Goole uses it.   

@MasterKale it seems like your use case is for and ideintier first flow where you know what transports are set on the available credentialID.    In that case, are you looking for the available transports as what you really want?  

Without an identifier first flow the RP should probably just do conditional mediation and let the OS sort it out.  

This sort of flag seems mostly useful in the case where the RP is doing identifier first with an allow list?

If the browser is going to say I support hybrid, or some combination of hybrid and platform is it still useful to you if the browser doesn't return no in the case BLE is disabled?

**Comment by MasterKale** [comment]
> @MasterKale it seems like your use case is for and ideintier first flow where you know what transports are set on the available credentialID. In that case, are you looking for the available transports as what you really want?

That's correct, we identify the user before populating `allowCredentials` with their credentials. I could see something that returns "available transports" as useful in troubleshooting user issues after failed authentication attempts. But if WebAuthn raises a "user clicked cancel" error for an allowCredentials with `["internal", "hybrid"]` in it because we were told both were possible but `"internal"` didn't match and the user gave up, we'd not really be better or worse off than we are now. Which might make "available transports" less useful than they first seem...

I'll take that back to my team and think it through. Maybe I can more coherently communicate my higher-level request in a new issue because I think it's probably just making PR feedback here too noisy.

**Comment by MasterKale** [comment]
> So ...what is a passkey?
To reiterate earlier discussion, and as per this PR, it's an alias for "discoverable credential". And in fact with FIDO Alliance shifting marketing of passkeys into "synced passkey" and "device-bound passkey" (so the synced part is no longer implied) shouldn't there be there should be fewer reasons not to make this connection.

> If you want to use this in the method, I would advise first defining what exactly is a passkey is. ...and pretty high up in the specs ...and also what a "passkey platform authenticator" means.
That's half of what this PR is trying to achieve. Look up "discoverable credential" under Terminology here and you'll see how this PR attempts to define "passkey":

https://pr-preview.s3.amazonaws.com/w3c/webauthn/1901/ad61c6f...0ad2938.html#sctn-terminology

> Wouldn't it be best to simply add isHybridAuthenticatorAvailable() to cover both cases?
Why not add both? `isPasskeyPlatformAuthenticatorAvailable()` gives the majority of RP developers a single method to call to check and see if passkeys are likely to be available for use. `isHybridAuthenticatorAvailable()` could then also be added for those RP's who want a bit more nuanced insights into platform capabilities with respect to handling hybrid auth. 

As I continue to say, let's make WebAuthn easier to use because it'll help with adoption.

**Comment by dagnelies** [comment]
> As I continue to say, let's make WebAuthn easier to use because it'll help with adoption.

Well, I'm all for that ^^ ...IMHO a big challenge is the large pool of unintuitive or unclear terminology.

When you see `isPasskeyPlatformAuthenticatorAvailable()` you basically have no idea what it does. It's impossible to guess. Exactly like `isConditionalMediationAvailable()` ...the first reaction is "What does that check?" and you spend more time figuring it out rather than using a more explicit method. ...Of course there are some tutorials that you can follow and people will just copy/paste ...but it's simply unclear due to the vocabulary.

If I would write it from scratch, I'd simply write it like this:

- `isPlatformAuthenticatorAvailable()` // with user verification assumed
- `isRoamingAuthenticatorAvailable()`
- `isAvailable()` // checks both


> To reiterate earlier discussion, and as per this PR, it's an alias for "discoverable credential". And in fact with FIDO Alliance shifting marketing of passkeys into "synced passkey" and "device-bound passkey" (so the synced part is no longer implied) shouldn't there be there should be fewer reasons not to make this connection.

See, you got me. I actually thought a "passkey" was a credential meant to be synced by the platform. I also always found this "discoverable" confusing.

I actually find "passkey" a nicer term than a "credential" because it's more clear. Also "synced" and "device-bound" really hit the spot since it's way more accurate than the current "backup" vocabulary.  Nevertheless, if passkey *is* a "discoverable credential", then this change should be applied to the whole spec and not add one more terminology alias. ...and if a passkey is a "discoverable credential", what is a "non-discoverable credential"? ;)

**Comment by ndpar** [comment]
I'm afraid it is too late to make passkey a synonym of discoverable credentials. There is no hint of discoverability in the [FIDO Alliance definition](https://fidoalliance.org/passkeys/)

> Based on FIDO standards, passkeys are a replacement for passwords that provide faster, easier, and more secure sign-ins to websites and apps across a user’s devices. Unlike passwords, passkeys are always strong and phishing-resistant.​
> Passkeys simplify account registration for apps and websites, are easy to use, work across most of a user’s devices, and even work on other devices within physical proximity.​

Of course you can ignore FIDO Alliance verbiage in W3C spec, but will it make adoption easier?

**Comment by Firstyear** [comment]
> That's half of what this PR is trying to achieve. Look up "discoverable credential" under Terminology here and you'll see how this PR attempts to define "passkey":
> 
> https://pr-preview.s3.amazonaws.com/w3c/webauthn/1901/ad61c6f...0ad2938.html#sctn-terminology

To reiterate from my earlier comment for @dagnelies there are no less than 4 definitions of passkey

* Any possible webauthn credentials (cite https://freeipa.readthedocs.io/en/latest/designs/passkeys.html )
* A synchronised/roaming credentials (cite https://support.okta.com/help/s/article/Passkey-Management?language=en_US )
* A credential that has BE/BS=true - I can't find an obvious cite for this, android could fall into this category though in a way.
* And the passkeys are resident keys as discussed here.

There is no consensus to what the term means, and including it in this specification does not assist here.

**Comment by dagnelies** [comment]
Agreed. Adding more vocabulary is confusing. One of the "adoption issues" currently out there is the confusing relationship between passkeys and webauthn.

**Comment by emlun** [comment]
There is consensus in the FIDO Alliance. The agreed-upon definition there (see https://fidoalliance.org/passkeys/#faq) is that a **passkey** is a _discoverable FIDO credential_, without further qualification:

>WHAT IS A PASSKEY?
>[...]
>From a technical standpoint, passkeys are “discoverable” FIDO credentials for passwordless authentication. [...]

That said, there still is clearly a lot of confusion about the term. I think one can argue both ways here: on the one hand, adding a "passkey" definition to WebAuthn might only confuse matters even more. On the other hand, repeating FIDO's "passkey" definition in WebAuthn could help spread and solidify that definition and help the wider community converge on one meaning of the term.

I'm not sure on which side I stand in that. Perhaps it is wise to avoid introducing the term. Do we really need it? I'm really not opposed to naming the method `isUserVerifyingPlatformAuthenticatorOrHybridTransportAvailable()` for example. Long names are only a problem if you need to type them all the time, which is very unlikely you'll need to do with this one.

**Comment by arianvp** [comment]
Just a thought, but how about we introduce a new function that give structured info about available credential type support?

```
getAvailableCredentialProperties : Promise<AvailableCredentialProperties>

interface AvailableCredentialProperties {
  transportsAvailable: []CredentialTransport
  conditionalMediationAvailable: bool
  userVerificationAvailable: bool
  residentKeyAvailable: bool
}
```

**Comment by dagnelies** [comment]
@emlun thanks for pointing out the definition. I think adding it to the spec would be a good think overall since it clarifies what it is and reduces confusion.

**Comment by Firstyear** [comment]
> There is consensus in the FIDO Alliance. The agreed-upon definition there (see https://fidoalliance.org/passkeys/#faq) is that a **passkey** is a _discoverable FIDO credential_, without further qualification:
> 

Given that there have been comments in this thread like https://github.com/w3c/webauthn/pull/1901#issuecomment-1583847785 it's hard to say that the FIDO alliance definition is being found and understood, especially going back to the fact that businesses are already deploying 'passkeys` using a variety of self-chosen definitions to what they are. 

I'm extremely against passkey as a term being introduced to this specification.

**Comment by ko-koiwai** [comment]
> > There is consensus in the FIDO Alliance. The agreed-upon definition there (see https://fidoalliance.org/passkeys/#faq) is that a **passkey** is a _discoverable FIDO credential_, without further qualification:

Yes I know that but it has been communicated differently within FIDO internally. (Please refer to the email from FIDO secretary dated May 11, 2023 if you subscribe to FIDO members mailing list.)

**Comment by nicksteele** [comment]
Unfortunately the term passkey has already been introduced to the standard indirectly, because passkeys are a type of WebAuthn credential and it's weird to treat them like we're saying Voldemort's name. Passkeys have existed for over a year, they are a relatively understood concept amongst this community, and in my opinion, to _avoid_ saying the word passkey would cause _more_ confusion and hinder any greater understanding of it by the dev community at large. Trying to not say 'passkey' by replacing it with `isUserVerifyingPlatformAuthenticatorOrHybridTransportAvailable` is just going to be far and away more confusing and less helpful than calling `isPasskey`

> there are no less than 4 definitions of passkey

Well that's a great reason to say the definition here, in the standard, from which passkeys are based. To @emlun's point we don't even need to define it ourselves, just link or reiterate FIDO's definition. I'd be fine adding a note to it about what it has in regards to spec terminology, but I'd say we have the most authority outside of the FIDO alliance to say what a passkey is or isn't

Passkeys are going to be the majority type of credential used with the WebAuthn API, to avoid the term because some other companies have already defined it incorrectly is cutting off the nose to spite the face.

File: index.bs
Comment (timcappalli) [reply index.bs:4319]: I'll address this in a separate PR (issue: #1912)

File: index.bs
Comment (timcappalli) [reply index.bs:4376]: addressed in [8813987](https://github.com/w3c/webauthn/pull/1901/commits/88139876d0418563ac8492f1eada3fc9eed4c9af)

**Review by pascoej — APPROVED** [review]

File: index.bs
Comment (timcappalli) [reply index.bs:4369]: Resolving the comment as there's been no activity in over a month

**Review by akshayku — APPROVED** [review]

**Review by agl — APPROVED** [review]

**Comment by timcappalli** [comment]
2023-07-12 call: 5 approvals, no objections to merging

**Comment by maxhata** [comment]
I think the term "passkeys" have strong implication on marketing.
There is a clear gap between this definition of passkeys and that of the WebAuthn partner organization, FIDO Alliance.
This will create further confusion on the issue.

**Comment by nadalin** [comment]
In the end the consensus of the WG determines what goes in the specification, I did not hear any objections here

**Comment by Firstyear** [comment]
> In the end the consensus of the WG determines what goes in the specification, I did not hear any objections here

I objected.

**Comment by maxhata** [comment]
I would object, too.

**Comment by dagnelies** [comment]
I objected too.

**Comment by nadalin** [comment]
@maxhata @Firstyear @dagnelies all the comments are reviewed before the merge and all merges are done by consensus on the bi-weekly calls, there were no objections, each bi-weekly call has an agenda posted and we discuss the issues  and pull requests listed, you are welcome to join the calls to further discuss. You are welcome to open a new issue and we can discuss on the next call. Since all merges are done on the calls the consensus is done there after review of GitHub comments.

**Comment by Firstyear** [comment]
@nadalin I can't attend due to timezones and my work schedule. 

It shouldn't be the case that people who are members don't have their voices heard because we can't join a call.

**Comment by nadalin** [comment]
I didn't say the voices weren't heard they said the decision is made on the call this has been the case for the last 5 years since the WG started, We review everything for doing the merge. All voices are heard whether you're on the call or not as long as you're comments are in the issue or pull request. We try to accommodate everybody on the call times as you know it always doesn't work out. So as I indicated if you don't agree with a pull request that was merged open up a new issue we have reversed pull requests in the past.

**Comment by devsnek** [comment]
So is this function trying to discourage people from using yubikeys? Is there some security issue with them? I'm very confused what the direction here is.

**Comment by MasterKale** [comment]
`isPasskeyPlatformAuthenticatorAvailable()` very specifically checks whether a **platform authenticator** capable of leveraging discoverable credentials is available, either locally or via the hybrid transport:

> [WebAuthn Relying Parties](https://w3c.github.io/webauthn/#webauthn-relying-party) use this method to determine whether they can create a new [passkey](https://w3c.github.io/webauthn/#passkey) using a [user-verifying platform authenticator](https://w3c.github.io/webauthn/#user-verifying-platform-authenticator) or a [hybrid](https://w3c.github.io/webauthn/#dom-authenticatortransport-hybrid) authenticator.

https://w3c.github.io/webauthn/#sctn-isPasskeyPlatformAuthenticatorAvailable

@devsnek To your question...

> What does this function return if i have a normal yubikey plugged in?

...`isPasskeyPlatformAuthenticatorAvailable()` returns the same thing that would be returned if no yubikey was plugged in. The presence of a non-hybrid roaming authenticator does not factor into this method's return value.

> False in either case. Is neither a platform authenticator or a hybrid transport.

This is incorrect. If a user-verifying platform authenticator is available and/or hybrid auth is possible then the method would return `true`. Again, the presence (or lack) of a security key does not change the availability of the platform authenticator or ability to use the hybrid transport.

**Comment by timcappalli** [comment]
Hey all - this is a merged / closed PR.

If you would like to propose changes to the spec, please either create a new issue or a new PR.

If you have questions, please use [FIDO-DEV](https://groups.google.com/a/fidoalliance.org/g/fido-dev) or the [Passkeys Developer discussion](https://github.com/passkeydeveloper/discussions/discussions).

**Pull request merged**
